src/uct/api/v2/uct_v2.h

```diff
@@ -1083,6 +1083,209 @@ ucs_status_t
 uct_rkey_compare(uct_component_h component, uct_rkey_t rkey1, uct_rkey_t rkey2,
                  const uct_rkey_compare_params_t *params, int *result);
 
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief @ref uct_iface_op_attr_t field mask flags
+ * The enumeration allows specifying which fields in @ref uct_iface_op_attr_t are
+ * present. It is used to enable backward compatibility support.
+ */
+typedef enum {
+    /** Enables @ref uct_iface_op_attr::max_short */
+    UCT_IFACE_OP_ATTR_FIELD_MAX_SHORT       = UCS_BIT(0),
+
+    /** Enables @ref uct_iface_op_attr::max_bcopy */
+    UCT_IFACE_OP_ATTR_FIELD_MAX_BCOPY       = UCS_BIT(1),
+
+    /** Enables @ref uct_iface_op_attr::min_zcopy */
+    UCT_IFACE_OP_ATTR_FIELD_MIN_ZCOPY       = UCS_BIT(2),
+
+    /** Enables @ref uct_iface_op_attr::max_zcopy */
+    UCT_IFACE_OP_ATTR_FIELD_MAX_ZCOPY       = UCS_BIT(3),
+
+    /** Enables @ref uct_iface_op_attr::opt_zcopy_align */
+    UCT_IFACE_OP_ATTR_FIELD_OPT_ZCOPY_ALIGN = UCS_BIT(4),
+
+    /** Enables @ref uct_iface_op_attr::align_mtu */
+    UCT_IFACE_OP_ATTR_FIELD_ALIGN_MTU       = UCS_BIT(5),
+
+    /** Enables @ref uct_iface_op_attr::max_zcopy_hdr */
+    UCT_IFACE_OP_ATTR_FIELD_MAX_ZCOPY_HDR   = UCS_BIT(6),
+
+    /** Enables @ref uct_iface_op_attr::max_iov */
+    UCT_IFACE_OP_ATTR_FIELD_MAX_IOV         = UCS_BIT(7),
+
+    /** Enables @ref uct_iface_op_attr::max_outstanding */
+    UCT_IFACE_OP_ATTR_FIELD_MAX_OUTSTANDING = UCS_BIT(8)
+} uct_iface_op_attr_field_t;
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Capabilities and limitations for a single operation on @ref uct_iface_t.
+ * 
+ * A structure that describes capabilities and limitations for short/bcopy/zcopy modes,
+ * as well as header sizes and alignment, for a single communication operation.
+ */
+typedef struct uct_iface_op_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_iface_op_attr_field_t.
+     */
+    uint64_t field_mask;
+    /** Maximal size for short */
+    size_t   max_short;
+    /** Maximal size for bcopy */
+    size_t   max_bcopy;
+    /** Minimal size for zcopy (total of
+     * @ref uct_iov_t::length of the @a iov parameter) */
+    size_t   min_zcopy;
+    /** Maximal size for zcopy (total of
+     * @ref uct_iov_t::length of the @a iov parameter) */
+    size_t   max_zcopy;
+    /** Optimal alignment for zero-copy buffer address */
+    size_t   opt_zcopy_align;
+    /** MTU used for alignment */
+    size_t   align_mtu;
+    /** Max. header size for zcopy */
+    size_t   max_zcopy_hdr;
+    /** Maximal @a iovcnt  */
+    size_t   max_iov;
+    /** Maximal number of simultaneous receive operations */
+    size_t   max_outstanding;
```

Comment:
maybe better to add to `uct_iface_attr_v2->tag` directly instead? (it is used only for hwtm receives)